timelock-agent/src/main/java/com/palantir/timelock/paxos/TimeLockAgent.java

```diff
@@ -65,24 +68,41 @@ public TimeLockAgent(TimeLockInstallConfiguration install,
         this.paxosResource = PaxosResource.create(install.paxos().dataDirectory().toString());
         this.leadershipCreator = new PaxosLeadershipCreator(install, runtime, registrar);
         this.lockCreator = new LockCreator(runtime, deprecated);
-        this.timestampCreator = new PaxosTimestampCreator(paxosResource,
-                PaxosRemotingUtils.getRemoteServerPaths(install),
-                PaxosRemotingUtils.getSslConfigurationOptional(install).map(SslSocketFactories::createSslSocketFactory),
-                JavaSuppliers.compose(TimeLockRuntimeConfiguration::paxos, runtime));
+        this.timestampCreator = getTimestampCreator();
         this.timelockCreator = install.asyncLock().useAsyncLockService()
                 ? new AsyncTimeLockServicesCreator(leadershipCreator, install.asyncLock())
                 : new LegacyTimeLockServicesCreator(leadershipCreator);
     }
 
+    private TimestampCreator getTimestampCreator() {
```

Comment:
should just have a method on the `TsBoundPersisterConfiguration` class for this, so you just do `install.timestampBoundPersistence().getTimestampCreator()` and avoid all the nasty reflection.

Reply:
Had considered that, but unfortunately `PaxosTsBoundPersisterConfiguration` does not have access to `PaxosResource`, runtime config etc. 

We can have something like `install.timestampBoundPersistence().getTimestampCreator(paxosResource,remoteServers, optionalSecurity, paxosRuntime)` but and ignore these parameters in `DatabaseTsBoundPersisterConfiguration`. Does that seem to be a better option?